src/targets/gpu/lowering.cpp

```diff
@@ -376,6 +377,26 @@ struct miopen_apply
             return mod->replace_instruction(ins, ins->get_operator(), inputs, ins->module_inputs());
         });
     }
+
+    /**
+     *  Adds reshape lazy to reshape ops that can be aliased instead of copied
+     */
+    void add_reshape_lazy_op()
+    {
+        apply_map.emplace("reshape", [=](instruction_ref ins) {
+            auto lazy_ins = mod->replace_instruction(
+                ins,
+                make_op("reshape_lazy", {{"dims", {ins->get_operator().to_value().at("dims")}}}),
+                ins->inputs(),
+                ins->module_inputs());
+
+            auto output2                       = insert_allocation(ins, ins->get_shape());
+            std::vector<instruction_ref> refs2 = ins->inputs();
+            refs2.push_back(output2);
+
+            return mod->insert_instruction(lazy_ins, make_op("gpu::contiguous"), refs2);
```

Comment:
I don't follow what is happening here. 

From what i understand this would make inputs of the "reshape" copy operation contiguous. 

But I dont' think it would pass them to `lazy_ins` because lazy_ins was inserted with original inputs.

Reply:
Yes, the replace_instruction should be done last.